core/lib/multivm/src/versions/vm_latest/tracers/refunds.rs

```diff
@@ -50,9 +51,10 @@ pub(crate) struct RefundsTracer {
     gas_spent_on_bytecodes_and_long_messages: u32,
     l1_batch: L1BatchEnv,
     pubdata_published: u32,
+    _phantom: PhantomData<S>,
 }
 
-impl RefundsTracer {
+impl<S> RefundsTracer<S> {
```

Comment:
Is this type argument required just for `dispatch_tracers!` to work out of the box in `impl Tracer for Tracer for DefaultExecutionTracer`? If so, IMO, that's a weird reason to add a type param to a type; `dispatch_tracers!` could be improved to support more complex expressions, or could be removed altogether (see the other comment).

Reply:
From the one side yes, but even though without this requirements it was an ugly trait specification for calls with memory..

Reply:
```
        <ResultTracer as DynTracer<S, SimpleMemory<H>>>::after_decoding(
            &mut self.result_tracer,
            state,
            data,
            memory,
        );

```
That's how it was before

Reply:
You can probably avoid specifying `SimpleMemory<H>` (i.e., use `DynTracer<S, _>`) in this particular case, which would simplify syntax a little. In general, I still think that using full method syntax is preferable to adding type params for a type in such cases; type params are appropriate if the type itself doesn't work without them, not if some method invocations require using full method syntax 🙃

Reply:
It actually simplifies the trait implementation.